specification/marketplacecatalog/resource-manager/Keys/stable/2023-01-01-preview/discoverykeys.json

```diff
@@ -0,0 +1,255 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Discovery API Keys Management",
+    "version": "2023-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/discovery/keys/create": {
+      "post": {
+        "tags": [
+          "CreateApiKey"
+        ],
+        "operationId": "CreateApiKey_Post",
+        "x-ms-examples": {
+          "Create_Key": {
+            "$ref": "./examples/CreateKey.json"
+          }
+        },
+        "parameters": [
+          {
+            "type": "string",
+            "name": "subscriptionId",
+            "in": "path",
+            "required": true,
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "x-ms-client-tenant-id",
+            "x-originalName": "tenantId",
+            "description": "The tenant that the subscription is in",
+            "in": "header",
+            "required": true,
+            "x-nullable": true
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "x-nullable": false,
+            "description": "Returns the newly created API key",
+            "schema": {
+              "$ref": "#/definitions/key"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/discovery/keys/revoke/{key}": {
```

Comment:
>discovery/keys/revoke/{key} [](http://example.com/codeflow?start=69&length=27)

This is also invalid, you need to follow ARM's RPC (linked pasted above), your API should be modeled as: /keys/{keyName}/{actionName}